### PR TITLE
Add (very basic) vg call

### PR DESCRIPTION
**Pull request by glennhickey**
vg call takes as input a vg pileup and produces an alignment for each snp in gam, which can be merged back into the graph with vg mod -i.  The logic is presently based off the samtools mpileup paper (but only supports single sample for time being).  This is a quick prototype for the purposes of our graph/mapping bake-off, but can hopefully help with baseline testing when more sophisticated approaches are implemented...

### CODE REVIEW

File: caller.hpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+#include "hash_map.hpp"
+#include "utility.hpp"
+#include "pileup.hpp"
+
+namespace vg {
+
+using namespace std;
+
+// Super simple variant caller, for now written to get bakeoff evaluation bootstrapped.
+// Idea: Idependently process Pileup records, using simple thresholding to make calls.
```
Comment (ekg) [inline caller.hpp]: You're no longer just thresholding though, no?

File: main.cpp [hunk 2]
```diff
@@ -17,12 +17,128 @@
...
+#include "caller.hpp"
 #include "google/protobuf/stubs/common.h"
 
 using namespace std;
 using namespace google::protobuf;
 using namespace vg;
 
+void help_call(char** argv) {
+    cerr << "usage: " << argv[0] << " call [options] <pileup.vgpu> > out.gam" << endl
+         << "Compute SNPs from pilup data (prototype! for evaluation only). " << endl
```
Comment (ekg) [inline main.cpp]: All of `vg` is for evaluation purposes only :smile:

File: caller.hpp (hunk 1 above)
Comment (glennhickey) [inline caller.hpp]: Oops, yeah, I added support for quality scores but didn't update that
comment (maybe others?).  Not sure if worth another pull request but happy
to do it..

Comment (ekg) [inline caller.hpp]: Nah we can fix it later.

**Pull request merged**